doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -706,6 +704,10 @@ has type :n:`@type`.
    is closed, the :n:`@ident`\(s) become undefined and every object depending on them will be explicitly
    parameterized (i.e., the variables are *discharged*).  See Section :ref:`section-mechanism`.
 
+   Assumptions that include the :n:`Inline` clause will be automatically
+   expanded when a functor is applied with :cmd:`Module`, except when the
+   functor application is prefixed by ``!``.
```

Comment:
Rather than duplicating this explanation here, where it is a bit out of context, I suggest saying something like: "the `Inline` clause is only relevant inside functors and described in <link to the module section>".